src/services/pacs_integration.tsx

```diff
@@ -0,0 +1,101 @@
+import { DcmImage, PFDCMResponse, PACSMainResponse, PACSSeries} from "../context/reducers/dicomImagesReducer";
+import { BASE_PACS_FILE_PATH } from '../api/app.config';
+import axios from 'axios';
+
+declare var process: {
+    env: {
+        REACT_APP_CHRIS_UI_PFDCM_URL: string,
+    }
+};
+
+class PACSIntegration {
+
+    static parseResponse(rawResponse: string): PFDCMResponse | undefined {
+        try {
+            let responseHeader = rawResponse.split('\r');
+            // the 4th element of responseHeader contains the stringified JSON body return payload
+            const responseBody = responseHeader?.[4];
+            const responseJSON: PFDCMResponse = JSON.parse(responseBody);
+            return responseJSON;
+        } catch (err) {
+            console.log(err);
+            return undefined;
+        }
+    }
+
+    /**
+     * Performs a PACS query on the given PatientID and returns an array of DcmImages containing metadata 
+     * of the DICOM files associated with that patient
+     * @param {string | undefined} patientID 
+     */
+    static async queryPatientFiles(patientID?: string): Promise<DcmImage[]> {
+        if (!patientID) return [];
+        
+        try {
+            const rawResponse = await axios.post(process.env.REACT_APP_CHRIS_UI_PFDCM_URL, JSON.stringify({
+                action:"PACSinteract",
+                meta:{
+                    do:"query",
+                    on:{PatientID: patientID},
+                    PACS:"orthanc"
+                }
+            }), {headers: {'Content-Type': 'text/plain'}});
+            const parsedResponse = this.parseResponse(rawResponse.data);
+            const data = parsedResponse?.query?.data;
+            const patientData = data ? data.flatMap((study: PACSMainResponse): DcmImage[] => (
+                study.series.map((series: PACSSeries): DcmImage => ({
+                        id: series.uid.value,
+                        creation_date: series.StudyDate.value,
+                        fname: BASE_PACS_FILE_PATH + series.SeriesInstanceUID.value,
+                        PatientID: series.PatientID.value,
+                        PatientName: series.PatientName.value,
+                        PatientBirthDate: series.PatientBirthDate.value,
+                        PatientAge: series.PatientAge.value,
+                        PatientSex: series.PatientSex.value,
+                        StudyInstanceUID: series.StudyInstanceUID.value,
+                        StudyDescription: series.StudyDescription.value,
+                        SeriesInstanceUID: series.SeriesInstanceUID.value,
+                        SeriesDescription: series.SeriesDescription.value,
+                        Modality: series.Modality.value,
+                        pacs_identifier: 'covidnet'
+                }))
+            )) : [];
+            
+            return patientData;
```

Comment:
For readability purposes, i would suggest moving the ternary conditional operation down to the return line.

Something along the lines of

```suggestion
            const patientData = data?.flatMap((study: PACSMainResponse): DcmImage[] => (
                study.series.map((series: PACSSeries): DcmImage => ({
                        id: series.uid.value,
                        creation_date: series.StudyDate.value,
                        fname: BASE_PACS_FILE_PATH + series.SeriesInstanceUID.value,
                        PatientID: series.PatientID.value,
                        PatientName: series.PatientName.value,
                        PatientBirthDate: series.PatientBirthDate.value,
                        PatientAge: series.PatientAge.value,
                        PatientSex: series.PatientSex.value,
                        StudyInstanceUID: series.StudyInstanceUID.value,
                        StudyDescription: series.StudyDescription.value,
                        SeriesInstanceUID: series.SeriesInstanceUID.value,
                        SeriesDescription: series.SeriesDescription.value,
                        Modality: series.Modality.value,
                        pacs_identifier: 'covidnet'
                }))
            ));
            
            return patientData ? patientData : [];
```

Reply:
Done https://github.com/darwinai/covidnet_ui/pull/8/commits/a9afeb39a2cf57df4f276e2df2371c40040229d1